src/plugins.c

```diff
@@ -1186,16 +1495,89 @@ static void pm_plugin_toggled(GtkCellRendererToggle *cell, gchar *pth, gpointer
 			keybindings_load_keyfile();		/* load shortcuts */
 
 		/* update model */
-		gtk_list_store_set(pm_widgets.store, &store_iter,
+		gtk_tree_store_set(pm_widgets.store, &store_iter,
 			PLUGIN_COLUMN_CHECK, state,
 			PLUGIN_COLUMN_PLUGIN, p, -1);
 
 		/* set again the sensitiveness of the configure and help buttons */
 		pm_update_buttons(p);
+
+		/* Depending on the state disable the checkbox for the proxy of this plugin, and
+		 * only re-enable if the proxy is not used by any other plugin */
+		if (p->proxy != &builtin_so_proxy_plugin)
+		{
+			GtkTreeIter parent;
+			gboolean can_uncheck;
+			GtkTreePath *store_path = gtk_tree_model_filter_convert_path_to_child_path(
+			                                GTK_TREE_MODEL_FILTER(model), path);
+
+			g_warn_if_fail(store_path != NULL);
+			if (gtk_tree_path_up(store_path))
+			{
+				gtk_tree_model_get_iter(GTK_TREE_MODEL(pm_widgets.store), &parent, store_path);
+
+				if (state)
+					can_uncheck = FALSE;
+				else
+					can_uncheck = p->proxy->proxied_count == 0;
+
+				gtk_tree_store_set(pm_widgets.store, &parent,
+					PLUGIN_COLUMN_CAN_UNCHECK, can_uncheck, -1);
+			}
+			gtk_tree_path_free(store_path);
+		}
+	}
+	/* We need to find out if a proxy was added or removed because that affects the plugin list
+	 * presented by the plugin manager */
+	if (prev_num_proxies != active_proxies.length)
+	{
+		/* Rescan the plugin list as we now support more. Gives some "already loaded" warnings
+		 * they are unproblematic */
+		if (prev_num_proxies < active_proxies.length)
+			load_all_plugins();
+
+		pm_populate(pm_widgets.store);
+		gtk_tree_view_expand_row(GTK_TREE_VIEW(pm_widgets.tree), path, FALSE);
 	}
+	gtk_tree_path_free(path);
 	g_free(file_name);
 }
 
+static void pm_populate(GtkTreeStore *store)
+{
+	GtkTreeIter iter;
+	GList *list;
+
+	gtk_tree_store_clear(store);
+	list = g_list_first(plugin_list);
+	if (list == NULL)
+	{
+		gtk_tree_store_append(store, &iter, NULL);
+		gtk_tree_store_set(store, &iter, PLUGIN_COLUMN_CHECK, FALSE,
+				PLUGIN_COLUMN_PLUGIN, NULL, -1);
+	}
+	else
+	{
+		Plugin *p;
+		for (; list != NULL; list = list->next)
+		{
+			p = list->data;
```

Comment:
Like for (Plugin *p = list->data; list != NULL; list = list->next, p = list ? list->data : NULL)?

I'm not sure that's an improvement really.